src/renderer/gdi/paint.cpp

```diff
@@ -525,6 +525,7 @@ bool GdiEngine::FontHasWesternScript(HDC hdc)
     wil::unique_hbrush hbr(CreateSolidBrush(gridlineColor));
     RETURN_HR_IF_NULL(E_FAIL, hbr.get());
 
+    // TODO: hbrPrev shouldn't be managed.
     wil::unique_hbrush hbrPrev(SelectBrush(_hdcMemoryContext, hbr.get()));
     RETURN_HR_IF_NULL(E_FAIL, hbrPrev.get());
     hbr.release(); // If SelectBrush was successful, GDI owns the brush. Release for now.

```

Comment:
@lhecker Can you have a look here?

Reply:
Nice find. And I agree with you. `wil` includes a helper to do this correctly so we can use something like this here:
```cpp
const auto restoreBrushOnExit = wil::SelectObject(_hdcMemoryContext, hbr.get());
```
This could replace all of the `hbrPrev` related code.

Reply:
Thanks for confirming. Also, `wil::SelectObject()` tip was helpful.